### PR TITLE
[BEAM-2956] Attempt to correctly report the Dataflow distribution in GCP related modules.

### PR DESCRIPTION
**Pull request by lukecwik**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by lukecwik** [comment]
R: @kennknowles

**Comment by lukecwik** [comment]
Note the corresponding DataflowJavaSDK change as well:
https://github.com/GoogleCloudPlatform/DataflowJavaSDK/pull/606

**Comment by lukecwik** [comment]
R: @tgroh

**Comment by lukecwik** [comment]
retest this please

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunnerInfo.java [hunk 1]
```diff
@@ -19,23 +19,66 @@
...
+        }
+      } catch (IOException e) {
+        LOG.warn("Error loading Dataflow runner properties resource: ", e);
+        properties = new Properties();
+      }
+
+      // Inherit the name and version from the Apache Beam distribution if this isn't
+      // the Dataflow distribution.
+      if (!properties.containsKey("name")) {
+        properties.setProperty("name", ReleaseInfo.getReleaseInfo().getName());
```
Comment (kennknowles) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunnerInfo.java]: Seems like this should be additive, not necessarily an overwrite.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java [hunk 2]
```diff
@@ -1081,7 +1084,20 @@ public BigtableWriteException(KV<ByteString, Iterable<Mutation>> record, Throwab
    * @see com.google.cloud.bigtable.config.BigtableVersionInfo
    */
   private static String getBeamSdkPartOfUserAgent() {
-    ReleaseInfo info = ReleaseInfo.getReleaseInfo();
+    // Attempt to get the Dataflow distribution release information, if we fail, fallback to using
+    // the Apache Beam release information.
+    ReleaseInfo info;
+    try {
+      Class<?> kls = ReflectHelpers.findClassLoader().loadClass(
+          "org.apache.beam.runners.dataflow.DataflowRunnerInfo");
```
Comment (kennknowles) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java]: This is gross, but I see why it has to happen, short term. Perhaps this is best delivered via PipelineOptions or some such, which can now be accessed via new DoFn.

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunnerInfo.java (hunk 1 above)
Comment (lukecwik) [reply runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunnerInfo.java]: Dataflow service currently relies on this property to detect deprecated SDKs

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java (hunk 2 above)
Comment (lukecwik) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java]: I'll revert this one as now I don't believe BigtableIO should care whether its Dataflow/Apache Beam distribution and if we need to patch BigtableIO, we can patch the user agent.

**Pull request closed without merging**